src/tabs/tabs.jsx

```diff
@@ -167,6 +224,134 @@ const Tabs = React.createClass({
       this.state.selectedIndex === index;
   },
 
+  _getDOMNode(refName) {
+    return ReactDOM.findDOMNode(this.refs[refName]);
+  },
+
+  _getDOMNodeWidth(refName) {
+    return this._getDOMNode(refName).offsetWidth;
+  },
+
+  _getSelectedTabWidth(tabIndex) {
+    return this._getDOMNode(Constants.TAB_ITEM_REF_NAME_PREFIX + tabIndex).offsetWidth;
+  },
+
+  _getSelectedTabLeftOffset(tabIndex) {
+    let tabLeftOffset = 0;
+    React.Children.forEach(this.props.children, (tab, index) => {
+      let tempWidth = this._getDOMNodeWidth(Constants.TAB_ITEM_REF_NAME_PREFIX + index);
+      if (index < tabIndex) {
+        tabLeftOffset += tempWidth;
+      }
+    });
+    return tabLeftOffset;
+  },
+
+  _handleLeftTabPaginatorTap() {
+    let tabContainerWidth = this._getDOMNodeWidth(Constants.TAB_CONTAINER_REF_NAME);
+    this._getDOMNode(Constants.TAB_SCROLL_WRAPPER_REF_NAME).scrollLeft -=
+      tabContainerWidth / this.getTabCount();
+    // tabContainerWidth needed due to that element might not have proper width
+    this.setState({
+      disableLeftPaginatorButton: this._disableLeftPaginatorButton(),
+      disableRightPaginatorButton: this._disableRightPaginatorButton(),
+      tabContainerWidth: tabContainerWidth,
+    });
+  },
+
+  _handleRightTabPaginatorTap() {
+    let tabContainerWidth = this._getDOMNodeWidth(Constants.TAB_CONTAINER_REF_NAME);
+    this._getDOMNode(Constants.TAB_SCROLL_WRAPPER_REF_NAME).scrollLeft +=
+      tabContainerWidth / this.getTabCount();
+    // tabContainerWidth needed due to that element might not have proper width
+    this.setState({
+      disableLeftPaginatorButton: this._disableLeftPaginatorButton(),
+      disableRightPaginatorButton: this._disableRightPaginatorButton(),
+      tabContainerWidth: tabContainerWidth,
+    });
+  },
+
+  _disableLeftPaginatorButton() {
+    return this._getDOMNode(Constants.TAB_SCROLL_WRAPPER_REF_NAME).scrollLeft === 0;
+  },
+
+
+  _disableRightPaginatorButton() {
+    let tabContainerWidth = this._getDOMNodeWidth(Constants.TAB_CONTAINER_REF_NAME);
+    let tabWrapperWidth = this._getDOMNodeWidth(Constants.TAB_WRAPPER_REF_NAME);
+    return this._getDOMNode(Constants.TAB_SCROLL_WRAPPER_REF_NAME).scrollLeft === tabContainerWidth - tabWrapperWidth;
+  },
+
+  handleWindowWidthChange() {
+    this.replaceState(this.getNewState());
```

Comment:
Can we not use `replaceState`?
> This method is not available on ES6 class components that extend React.Component. It may be removed entirely in a future version of React.